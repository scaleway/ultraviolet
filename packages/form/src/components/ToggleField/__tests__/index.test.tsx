import { screen } from '@testing-library/react'
import { renderWithForm } from '@utils/test'
import { describe, expect, test } from 'vitest'
import { ToggleField } from '..'

describe('ToggleField', () => {
  test('should render correctly', () => {
    const { asFragment } = renderWithForm(<ToggleField name="test" />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('should render correctly disabled', () => {
    const { asFragment } = renderWithForm(<ToggleField disabled name="test" />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('should render correctly checked', () => {
    const { asFragment } = renderWithForm(<ToggleField name="test" />, {
      defaultValues: {
        test: true,
      },
    })
    const element = screen.getByRole<HTMLInputElement>('checkbox')
    expect(element.checked).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })

  test('should render correctly with label and checked', () => {
    const { asFragment } = renderWithForm(
      <ToggleField label="test" name="test" />,
      {
        defaultValues: {
          test: true,
        },
      },
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
