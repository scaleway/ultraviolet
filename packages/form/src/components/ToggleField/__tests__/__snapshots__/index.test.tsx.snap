// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ToggleField > should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.emotion-0:active .emotion-3[data-disabled='false']:after {
  width: calc(1rem * 1.3775);
}

.emotion-0[aria-disabled="true"] {
  cursor: not-allowed;
  color: #b5b7bd;
}

.emotion-2 {
  box-sizing: content-box;
  outline: none;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  border-radius: 1rem;
  position: relative;
  -webkit-transition: all 300ms;
  transition: all 300ms;
  background-color: #e9eaeb;
  width: 3rem;
  height: 1.5rem;
}

.emotion-2:hover {
  background-color: #d9dadd;
}

.emotion-2:after {
  content: "";
  position: absolute;
  top: calc(1.5rem / 2 - 1rem / 2);
  left: 5px;
  width: 1rem;
  height: 1rem;
  border-radius: 100%;
  background-color: #ffffff;
  -webkit-transition: all 300ms;
  transition: all 300ms;
}

.emotion-2:focus-within,
.emotion-2:focus {
  box-shadow: 0px 0px 0px 3px #151a2d5c;
}

.emotion-2[data-disabled="false"]:active:after {
  width: calc(1rem * 1.3775);
}

.emotion-2:has(:checked) {
  color: #222638;
  background-color: #8c40ef;
}

.emotion-2:has(:checked):hover {
  background-color: #792dd4;
}

.emotion-2:has(:checked):after {
  left: calc(100% - 5px);
  -webkit-transform: translateX(-100%);
  -moz-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

.emotion-2:has(:checked):focus-within,
.emotion-2:has(:checked):focus {
  box-shadow: 0px 0px 0px 3px #8c40ef40;
}

.emotion-2[data-disabled="true"] {
  background: #f3f3f4;
}

.emotion-2[data-disabled="true"]:has(:checked) {
  background: #e5dbfd;
}

.emotion-2[data-error="true"] {
  background-color: #ffebf2;
}

.emotion-2[data-error="true"]:focus-within,
.emotion-2[data-error="true"]:focus {
  box-shadow: 0px 0px 0px 3px #f91b6c40;
}

.emotion-2[data-error="true"]:has(:checked) {
  background-color: #e51963;
}

.emotion-2[data-error="true"][data-disabled="true"] {
  background-color: #fff3f7;
}

.emotion-2[data-error="true"][data-disabled="true"]:has(:checked) {
  background-color: #ffd3e3;
}

.emotion-4 {
  position: absolute;
  opacity: 0;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  cursor: pointer;
}

.emotion-4[disabled] {
  cursor: not-allowed;
}

<div
    data-testid="testing"
  >
    <form
      novalidate=""
      style="display: contents;"
    >
      <label
        aria-disabled="false"
        class="emotion-0 emotion-1"
      >
        <div
          class="uv_toi52u0 uv_toi52u4j uv_toi52u2d uv_toi52u7d uv_toi52u21 uv_toi52u5d"
          style="--uv_4k0ekn0: var(--uv_4k0ekn0); --uv_4k0ekn1: var(--uv_4k0ekn1); --uv_4k0ekn2: var(--uv_4k0ekn2); --uv_4k0ekn3: var(--uv_4k0ekn3);"
        />
        <div
          class="emotion-2 emotion-3"
          data-disabled="false"
          data-error="false"
        >
          <input
            aria-invalid="false"
            class="emotion-4 emotion-5"
            id="«r0»"
            name="test"
            type="checkbox"
          />
        </div>
      </label>
    </form>
  </div>
</DocumentFragment>
`;

exports[`ToggleField > should render correctly checked 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.emotion-0:active .emotion-3[data-disabled='false']:after {
  width: calc(1rem * 1.3775);
}

.emotion-0[aria-disabled="true"] {
  cursor: not-allowed;
  color: #b5b7bd;
}

.emotion-2 {
  box-sizing: content-box;
  outline: none;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  border-radius: 1rem;
  position: relative;
  -webkit-transition: all 300ms;
  transition: all 300ms;
  background-color: #e9eaeb;
  width: 3rem;
  height: 1.5rem;
}

.emotion-2:hover {
  background-color: #d9dadd;
}

.emotion-2:after {
  content: "";
  position: absolute;
  top: calc(1.5rem / 2 - 1rem / 2);
  left: 5px;
  width: 1rem;
  height: 1rem;
  border-radius: 100%;
  background-color: #ffffff;
  -webkit-transition: all 300ms;
  transition: all 300ms;
}

.emotion-2:focus-within,
.emotion-2:focus {
  box-shadow: 0px 0px 0px 3px #151a2d5c;
}

.emotion-2[data-disabled="false"]:active:after {
  width: calc(1rem * 1.3775);
}

.emotion-2:has(:checked) {
  color: #222638;
  background-color: #8c40ef;
}

.emotion-2:has(:checked):hover {
  background-color: #792dd4;
}

.emotion-2:has(:checked):after {
  left: calc(100% - 5px);
  -webkit-transform: translateX(-100%);
  -moz-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

.emotion-2:has(:checked):focus-within,
.emotion-2:has(:checked):focus {
  box-shadow: 0px 0px 0px 3px #8c40ef40;
}

.emotion-2[data-disabled="true"] {
  background: #f3f3f4;
}

.emotion-2[data-disabled="true"]:has(:checked) {
  background: #e5dbfd;
}

.emotion-2[data-error="true"] {
  background-color: #ffebf2;
}

.emotion-2[data-error="true"]:focus-within,
.emotion-2[data-error="true"]:focus {
  box-shadow: 0px 0px 0px 3px #f91b6c40;
}

.emotion-2[data-error="true"]:has(:checked) {
  background-color: #e51963;
}

.emotion-2[data-error="true"][data-disabled="true"] {
  background-color: #fff3f7;
}

.emotion-2[data-error="true"][data-disabled="true"]:has(:checked) {
  background-color: #ffd3e3;
}

.emotion-4 {
  position: absolute;
  opacity: 0;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  cursor: pointer;
}

.emotion-4[disabled] {
  cursor: not-allowed;
}

<div
    data-testid="testing"
  >
    <form
      novalidate=""
      style="display: contents;"
    >
      <label
        aria-disabled="false"
        class="emotion-0 emotion-1"
      >
        <div
          class="uv_toi52u0 uv_toi52u4j uv_toi52u2d uv_toi52u7d uv_toi52u21 uv_toi52u5d"
          style="--uv_4k0ekn0: var(--uv_4k0ekn0); --uv_4k0ekn1: var(--uv_4k0ekn1); --uv_4k0ekn2: var(--uv_4k0ekn2); --uv_4k0ekn3: var(--uv_4k0ekn3);"
        />
        <div
          class="emotion-2 emotion-3"
          data-disabled="false"
          data-error="false"
        >
          <input
            aria-invalid="false"
            checked=""
            class="emotion-4 emotion-5"
            id="«r4»"
            name="test"
            type="checkbox"
          />
        </div>
      </label>
    </form>
  </div>
</DocumentFragment>
`;

exports[`ToggleField > should render correctly disabled 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.emotion-0:active .emotion-3[data-disabled='false']:after {
  width: calc(1rem * 1.3775);
}

.emotion-0[aria-disabled="true"] {
  cursor: not-allowed;
  color: #b5b7bd;
}

.emotion-2 {
  box-sizing: content-box;
  outline: none;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  border-radius: 1rem;
  position: relative;
  -webkit-transition: all 300ms;
  transition: all 300ms;
  background-color: #e9eaeb;
  width: 3rem;
  height: 1.5rem;
}

.emotion-2:hover {
  background-color: #d9dadd;
}

.emotion-2:after {
  content: "";
  position: absolute;
  top: calc(1.5rem / 2 - 1rem / 2);
  left: 5px;
  width: 1rem;
  height: 1rem;
  border-radius: 100%;
  background-color: #ffffff;
  -webkit-transition: all 300ms;
  transition: all 300ms;
}

.emotion-2:focus-within,
.emotion-2:focus {
  box-shadow: 0px 0px 0px 3px #151a2d5c;
}

.emotion-2[data-disabled="false"]:active:after {
  width: calc(1rem * 1.3775);
}

.emotion-2:has(:checked) {
  color: #222638;
  background-color: #8c40ef;
}

.emotion-2:has(:checked):hover {
  background-color: #792dd4;
}

.emotion-2:has(:checked):after {
  left: calc(100% - 5px);
  -webkit-transform: translateX(-100%);
  -moz-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

.emotion-2:has(:checked):focus-within,
.emotion-2:has(:checked):focus {
  box-shadow: 0px 0px 0px 3px #8c40ef40;
}

.emotion-2[data-disabled="true"] {
  background: #f3f3f4;
}

.emotion-2[data-disabled="true"]:has(:checked) {
  background: #e5dbfd;
}

.emotion-2[data-error="true"] {
  background-color: #ffebf2;
}

.emotion-2[data-error="true"]:focus-within,
.emotion-2[data-error="true"]:focus {
  box-shadow: 0px 0px 0px 3px #f91b6c40;
}

.emotion-2[data-error="true"]:has(:checked) {
  background-color: #e51963;
}

.emotion-2[data-error="true"][data-disabled="true"] {
  background-color: #fff3f7;
}

.emotion-2[data-error="true"][data-disabled="true"]:has(:checked) {
  background-color: #ffd3e3;
}

.emotion-4 {
  position: absolute;
  opacity: 0;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  cursor: pointer;
}

.emotion-4[disabled] {
  cursor: not-allowed;
}

<div
    data-testid="testing"
  >
    <form
      novalidate=""
      style="display: contents;"
    >
      <label
        aria-disabled="true"
        class="emotion-0 emotion-1"
      >
        <div
          class="uv_toi52u0 uv_toi52u4j uv_toi52u2d uv_toi52u7d uv_toi52u21 uv_toi52u5d"
          style="--uv_4k0ekn0: var(--uv_4k0ekn0); --uv_4k0ekn1: var(--uv_4k0ekn1); --uv_4k0ekn2: var(--uv_4k0ekn2); --uv_4k0ekn3: var(--uv_4k0ekn3);"
        />
        <div
          class="emotion-2 emotion-3"
          data-disabled="true"
          data-error="false"
        >
          <input
            aria-invalid="false"
            class="emotion-4 emotion-5"
            disabled=""
            id="«r2»"
            name="test"
            type="checkbox"
          />
        </div>
      </label>
    </form>
  </div>
</DocumentFragment>
`;

exports[`ToggleField > should render correctly with label and checked 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.emotion-0:active .emotion-3[data-disabled='false']:after {
  width: calc(1rem * 1.3775);
}

.emotion-0[aria-disabled="true"] {
  cursor: not-allowed;
  color: #b5b7bd;
}

.emotion-2 {
  box-sizing: content-box;
  outline: none;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  border-radius: 1rem;
  position: relative;
  -webkit-transition: all 300ms;
  transition: all 300ms;
  background-color: #e9eaeb;
  width: 3rem;
  height: 1.5rem;
}

.emotion-2:hover {
  background-color: #d9dadd;
}

.emotion-2:after {
  content: "";
  position: absolute;
  top: calc(1.5rem / 2 - 1rem / 2);
  left: 5px;
  width: 1rem;
  height: 1rem;
  border-radius: 100%;
  background-color: #ffffff;
  -webkit-transition: all 300ms;
  transition: all 300ms;
}

.emotion-2:focus-within,
.emotion-2:focus {
  box-shadow: 0px 0px 0px 3px #151a2d5c;
}

.emotion-2[data-disabled="false"]:active:after {
  width: calc(1rem * 1.3775);
}

.emotion-2:has(:checked) {
  color: #222638;
  background-color: #8c40ef;
}

.emotion-2:has(:checked):hover {
  background-color: #792dd4;
}

.emotion-2:has(:checked):after {
  left: calc(100% - 5px);
  -webkit-transform: translateX(-100%);
  -moz-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

.emotion-2:has(:checked):focus-within,
.emotion-2:has(:checked):focus {
  box-shadow: 0px 0px 0px 3px #8c40ef40;
}

.emotion-2[data-disabled="true"] {
  background: #f3f3f4;
}

.emotion-2[data-disabled="true"]:has(:checked) {
  background: #e5dbfd;
}

.emotion-2[data-error="true"] {
  background-color: #ffebf2;
}

.emotion-2[data-error="true"]:focus-within,
.emotion-2[data-error="true"]:focus {
  box-shadow: 0px 0px 0px 3px #f91b6c40;
}

.emotion-2[data-error="true"]:has(:checked) {
  background-color: #e51963;
}

.emotion-2[data-error="true"][data-disabled="true"] {
  background-color: #fff3f7;
}

.emotion-2[data-error="true"][data-disabled="true"]:has(:checked) {
  background-color: #ffd3e3;
}

.emotion-4 {
  position: absolute;
  opacity: 0;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  cursor: pointer;
}

.emotion-4[disabled] {
  cursor: not-allowed;
}

<div
    data-testid="testing"
  >
    <form
      novalidate=""
      style="display: contents;"
    >
      <label
        aria-disabled="false"
        class="emotion-0 emotion-1"
      >
        <div
          class="uv_toi52u0 uv_toi52u4j uv_toi52u2d uv_toi52u7d uv_toi52u21 uv_toi52u5d"
          style="--uv_4k0ekn0: var(--uv_4k0ekn0); --uv_4k0ekn1: var(--uv_4k0ekn1); --uv_4k0ekn2: var(--uv_4k0ekn2); --uv_4k0ekn3: var(--uv_4k0ekn3);"
        >
          <div
            class="uv_x6hyh50 uv_x6hyh52p uv_x6hyh57"
            style="--uv_3mg8xe0: auto 1fr; --uv_3mg8xe1: auto 1fr; --uv_3mg8xe2: auto 1fr; --uv_3mg8xe3: auto 1fr; --uv_3mg8xe4: auto 1fr; --uv_3mg8xe5: auto 1fr;"
          >
            <span
              class="uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owb uv_m4c9owi uv_m4c9owm uv_m4c9ow1b uv_m4c9ow20 uv_m4c9ow3c"
            >
              test
            </span>
          </div>
        </div>
        <div
          class="emotion-2 emotion-3"
          data-disabled="false"
          data-error="false"
        >
          <input
            aria-invalid="false"
            checked=""
            class="emotion-4 emotion-5"
            id="«r6»"
            name="test"
            type="checkbox"
          />
        </div>
      </label>
    </form>
  </div>
</DocumentFragment>
`;
